lib/emque/consuming/adapters/rabbit_mq/worker.rb

```diff
@@ -73,11 +73,33 @@ def process_message(delivery_info, metadata, payload)
               )
               ::Emque::Consuming::Consumer.new.consume(:process, message)
               channel.ack(delivery_info.delivery_tag)
-            rescue StandardError
-              channel.nack(delivery_info.delivery_tag)
+            rescue StandardError => ex
+              if retryable_errors.any? { |error| ex.class.to_s =~ /#{error}/ }
+                headers = metadata[:headers] || {}
+                retry_count = headers.fetch("x-retry-count", 0)
+
+                if retry_count <= retryable_error_limit
+                  logger.info("Retrying Retryable Error: #{ex.class}, with count #{retry_count}")
+                  headers["x-retry-count"] = retry_count + 1
+                  queue.publish(payload, {:headers =>headers})
+                else
+                  logger.info("Retryable Error: #{ex.class} ran out of retires at #{retry_count}, dropping message.")
```

Comment:
"dropping" may be confusing, as it denotes a message that cannot be processed again?